extractor/src/main/java/org/schabi/newpipe/extractor/utils/ManifestCreatorCache.java

```diff
@@ -0,0 +1,303 @@
+        return clearFactor;
+    }
+
+    /**
+     * Sets the clear factor of the cache, used when the cache limit size is reached.
+     *
+     * <p>
+     * The clear factor must be a double between {@code 0} excluded and {@code 1} excluded.
+     * </p>
+     *
+     * <p>
+     * Note that it will be only used the next time the cache size limit is reached.
+     * </p>
+     *
+     * @param clearFactor the new clear factor of the cache
+     * @throws IllegalArgumentException if the clear factor passed a parameter is invalid
+     */
+    public void setClearFactor(final double clearFactor) {
+        if (clearFactor <= 0 || clearFactor >= 1) {
+            throw new IllegalArgumentException("Invalid clear factor");
+        }
+
+        this.clearFactor = clearFactor;
+    }
+
+    /**
+     * Resets the clear factor to its {@link #DEFAULT_CLEAR_FACTOR default value}.
+     */
+    public void resetClearFactor() {
+        this.clearFactor = DEFAULT_CLEAR_FACTOR;
+    }
+
+    /**
+     * Reveals whether an object is equal to a {@code ManifestCreator} cache existing object.
+     *
+     * @param obj the object to compare with the current {@code ManifestCreatorCache} object
+     * @return whether the object compared is equal to the current {@code ManifestCreatorCache}
+     * object
+     */
+    @Override
+    public boolean equals(final Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+
+        final ManifestCreatorCache<?, ?> manifestCreatorCache =
+                (ManifestCreatorCache<?, ?>) obj;
+        return maximumSize == manifestCreatorCache.maximumSize
+                && Double.compare(manifestCreatorCache.clearFactor, clearFactor) == 0
+                && concurrentHashMap.equals(manifestCreatorCache.concurrentHashMap);
+    }
+
+    /**
+     * Returns a hash code of the current {@code ManifestCreatorCache}, using its
+     * {@link #maximumSize maximum size}, {@link #clearFactor clear factor} and
+     * {@link #concurrentHashMap internal concurrent hash map} used as a cache.
+     *
+     * @return a hash code of the current {@code ManifestCreatorCache}
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hash(maximumSize, clearFactor, concurrentHashMap);
+    }
+
+    /**
+     * Returns a string version of the {@link ConcurrentHashMap} used internally as the cache.
+     *
+     * @return the string version of the {@link ConcurrentHashMap} used internally as the cache
+     */
+    @Override
+    public String toString() {
+        return concurrentHashMap.toString();
+    }
+
+    /**
+     * Keeps only the newest entries in a cache.
+     *
+     * <p>
+     * This method will first collect the entries to remove by looping through the concurrent hash
+     * map
+     * </p>
+     *
+     * @param newLimit the new limit of the cache
```

Comment:
Personal opinion ahead, **do not make any changes**. Maybe I should create a separate issue, but this is not really a big problem, just wanted to open a small discussion.

I don't like this style of javadocs: they have html, they span too many lines and they have empty newlines in between. I'd prefer this:
```suggestion
     * Keeps only the newest entries in a cache. Will first collect the entries to remove by looping
     * through the concurrent hash map
     * @param newLimit the new limit of the cache
```
My suggestion uses 3 lines instead of 8, making it much easier to read when looking at the code. I know I can just have Android Studio show me a nice description that replaces the actual `/**/` comment, but that's not possible when reviewing or editing.
I don't think having a `<p>` in javadocs is much useful, since javadocs are usually pretty small, so there is no need to add separators. Also, if the useful information is in the first sentence, anybody reading the documentation can just stop at the first dot and they should already have a pretty good understanding, they do not need a `<p>` to know they should not proceed imho. I would only approve of html tags in javadocs if the javadocs were really long and/or had to describe a complex subject, but this does not seem the case in these functions.
Removing empty newlines makes it simpler to scroll through code by seeing compact blocks of documentation followed by a method, instead of some lines separated here and there that to the eye seem to be different things to analyze separately.